jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/AdministrationControllerTest.groovy

```diff
@@ -161,6 +164,19 @@ class AdministrationControllerTest extends Specification {
             notEditable << [ADMIN, USER, BANNED_USER]
     }
 
+    def 'must not be able to register with e-mail from blacklist'(){
```

Comment:
Since you duplicated the functionality in both AJAX and form registration you'd also need to have 2 tests for both of them.